### ISSUE TITLE
allow bsb to be run from globally installed bs-platform

### ISSUE THREAD

**Issue by jonni-larjomaa** [jaredly/reason-language-server#393]
bs-platform related binaries are by default expected to reside in the projects node_modules folder. allow changing this value to something else for example globally installed bs-platform support.

**Comment by jaredly** [comment]
Does globally installed bs-platform work these days? It never actually was able to build for me, so I haven't bothered supporting it.

**Comment by cristianoc** [comment]
It's definitely not recommended. OK when a new version of bs comes out every year or so, at most, which is not true at the moment.
Like for the build, these days each project specifies their own version of bs.

**Comment by jonni-larjomaa** [comment]
yes globally installed bsb works for me atleast? on mac os x catalina and installed with npm i -g bs-platform. The problem with project specific approach is tied to running dev env inside docker-containers, the binaries are not compatible between linux / darwin envs. and running vscode IDE natively in mac and the whole other environment (with project mounted volumes) inside docker. the ide cannot run the project specific bsb as it is build with linux kernel. as for globally installed would be build for mac. and also for the reason that ides would need project specific stuff running in order to help with compiling / type hinting etc. is bad design IMHO.

**Comment by jonni-larjomaa** [comment]
Just wondering why setting the bsb binary path to something else than default would harm anyone? by a configuration value.

**Comment by cristianoc** [comment]
@jonni-larjomaa leaving the IDE aside for a second. Where does the build take place, inside the linux container or outside on the Mac?

**Comment by jonni-larjomaa** [comment]
@cristianoc Basically linux-container does not need anything else than compiled js, so build could happen in mac, but. here is the but, npm install is run inside linux-container this is because all depencies that do dynamic compilation of binary functions needs to happen inside linux-container. So how this is accomplished is basically mounting the project folder inside docker. and i don't want to go to the extremes of "exclude bs-platform and depencies from mount" and then separetly running npm install inside docker and on mac (which would still not solve the problem of the IDE!).
